src/olympia/landfill/tests/test_stats.py

```diff
@@ -0,0 +1,39 @@
+from olympia import amo
+from olympia.addons.models import Addon
+from olympia.amo.tests import TestCase
+from olympia.stats.models import DownloadCount, UpdateCount
+from olympia.landfill.stats import (
+    generate_download_counts,
+    generate_update_counts,
+)
+
+
+class TestGenerateDownloadCounts(TestCase):
+    def setUp(self):
+        super().setUp()
+
+        self.addon = Addon.objects.create(type=amo.ADDON_EXTENSION)
+
+    def test_generate_download_counts(self):
+        x_days = 10
+
+        generate_download_counts(self.addon, x_days)
+
+        assert DownloadCount.objects.all().count() == x_days
+
+
+class TestGenerateUpdateCounts(TestCase):
+    def setUp(self):
+        super().setUp()
+
+        self.addon = Addon.objects.create(type=amo.ADDON_EXTENSION)
+
+    def test_generate_update_counts(self):
+        x_days = 10
+
+        generate_update_counts(self.addon, x_days)
+
+        assert UpdateCount.objects.all().count() == x_days
+        update_count = UpdateCount.objects.all()[0]
+        assert 'Darwin' in update_count.oses
```

Comment:
Just wondering why you decided to test these two things, but not `applications`, nor `sources` in the test for `generate_download_counts`?

Reply:
ha, that's because I added the `sources` after I wrote the test cases.. I added some more assertions. These tests don't really need to be ultra exhaustive so only a few things are checked.